lib/dnsimple/client/certificates.rb

```diff
@@ -108,6 +108,158 @@ def certificate_private_key(account_id, domain_id, certificate_id, options = {})
         Dnsimple::Response.new(response, Struct::CertificateBundle.new(response["data"]))
       end
 
+      # Purchase a Let's Encrypt certificate.
+      #
+      # @see https://developer.dnsimple.com/v2/domains/certificates/#letsencrypt-purchase
+      #
+      # @param  [Integer] account_id the account ID
+      # @param  [#to_s] domain_id the domain ID or domain name
+      # @param  [Hash] attributes
+      # @option attributes [Integer] :contact_id the contact ID (mandatory)
+      # @option attributes [String] :name the certificate name (optional)
```

Comment:
AFAIK the name should be mandatory (as we do for the record). If for the root domain, it should  be an empty string `""`.